### ISSUE TITLE
[package] Boost: when python is enabled the path to the interpreter gets stored in the recipe which causes issues.

### ISSUE THREAD

**Issue by ericriff** [conan-io/conan-center-index#10918]
### Package and Environment Details (include every applicable attribute)
  * Package Name/Version: **boost/x**
  * Operating System+version: **Linux Ubuntu 20.04**
  * Compiler+version: **GCC 8**
  * Docker image: **conanio/gcc8**
  * Conan version: **conan 1.48.1**
  * Python version: **Python 3.8.10**

By default `boost` gets built without python support, maybe that's why this hasn't come up yet. When python is enabled, there is some code that checks the interpreter path and stores it as a conan option (`python_executable`). Even though we delete this information from the package id (https://github.com/conan-io/conan-center-index/blob/master/recipes/boost/all/conanfile.py#L548) it still ends up in the package metadata. This becomes a problem when the package gets built on a given environment but then it is consumed somewhere else, where the python interpreter is located somewhere else.
This is exactly the kind of issue I'm seeing, mainly because we create the package on a disposable virtual env, so the path to the python interpreter changes frequently.
 

### Steps to reproduce (Include if Applicable)

This is a reference `conanfile`:

```py
from conans import ConanFile

class ConanError(ConanFile):
    name = "sample"
    version = "1.0.0"
    settings = "arch", "build_type"


    def requirements(self):
        self.requires("boost/1.79.0")
        self.options["boost"].without_python = False
```

Create a virtual env to simulate the build environment. We install `conan` and `numpy` in it. Why `numpy`? The recipe errors out otherwise, but that's a story for another ticket https://github.com/conan-io/conan-center-index/issues/8826#issue-1099315179.
```bash
virtualenv .venv_build --python 3.8
source .venv_build/bin/activate
pip install conan numpy
```

We can now build the package:
```bash
conan install ./conanfile.py --build=missing
```
If you capture this dependency tree on a lockfile, you'll get the first clue that something is wrong:
```bash
conan install ./conanfile.py --lockfile conan.lock
```
Inside that lockfile you can find the following information:
```bash
python_executable=/home/eric/conanLockfilesError/.venv_build/bin/python
```
Which will prevent the consumption of these package on an environment where the interpreter is located somewhere else. To replicate this, we delete the build environment and create a new one:
```bash
rm -rf .venv_build/
virtualenv .venv_consume --python 3.8
source .venv_consume/bin/activate
pip install conan
```
Now if we try to install the packages using:
```bash
conan install ./conanfile.py --lockfile conan.lock
```

Conan errors out because it tries to use the python interpreter stored in the package metadata, which path exists only on the build environment.

```bash
Using lockfile: '/home/eric/conanLockfilesError/conan.lock'
Configuration:
[settings]
arch=x86_64
arch_build=x86_64
build_type=Release
compiler=gcc
compiler.libcxx=libstdc++
compiler.version=8
os=Linux
os_build=Linux
[options]
[build_requires]
[env]

boost/1.79.0: running "/home/eric/conanLockfilesError/.venv_build/bin/python" -c "from __future__ import print_function; import sys; print('{}.{}'.format(sys.version_info[0], sys.version_info[1]))"
boost/1.79.0: (failed)
ERROR: boost/1.79.0: Invalid configuration: detected python version None doesn't match conan option 3.8

```
I think a valid solution would be to remove the `python_executable` option and replace it with a `property` or something similar, that gets populated when we run the recipe. At the end of the day we don't care about where the interpreter is, we only care about the version.

**Comment by SSE4** [comment]
might be a conan client bug (https://github.com/conan-io/conan/issues/11137), which is going to be never fixed.
we may try to add some workaround (e.g. use special value for `system` interpreter and evaluate it later, so it's never locked).

**Comment by hanusek** [comment]
@ericriff Have you solved it?

**Comment by ericriff** [comment]
I ended up hacking the recipe on our internal mirror (we host our own conan index / conan remote) to replace the `option` with a `property`.

**Comment by rudi-cilibrasi** [comment]
@ericriff  or anybody else able to share more details about the option -> property workaround? we are blocked on this issue as well currently
